scripts/_helpers.py

```diff
@@ -1200,118 +1200,83 @@ def download_GADM(country_code, update=False, out_logging=False):
-        ]
-
-        # create a subindex column that is useful
-        # in the GADM processing of sub-national zones
-        geodf_temp["GADM_ID"] = geodf_temp[f"GID_{code_layer}"]
+def _get_shape_col_gdf(path_to_gadm, co, gadm_level, gadm_clustering):
+    from build_shapes import get_GADM_layer
 
-        # concatenate geodataframes
-        geodf_list = pd.concat([geodf_list, geodf_temp])
-
-    geodf_GADM = gpd.GeoDataFrame(pd.concat(geodf_list, ignore_index=True))
-    geodf_GADM.set_crs(geodf_list[0].crs, inplace=True)
+    col = "name"
+    if not gadm_clustering:
+        gdf_shapes = gpd.read_file(path_to_gadm)
+    else:
+        if path_to_gadm:
+            gdf_shapes = gpd.read_file(path_to_gadm)
+            if "GADM_ID" in gdf_shapes.columns:
+                col = "GADM_ID"
 
-    return geodf_GADM
+                if gdf_shapes[col][0][
+                    :3
+                ].isalpha():  # TODO clean later by changing all codes to 2 letters
+                    gdf_shapes[col] = gdf_shapes[col].apply(
+                        lambda name: three_2_two_digits_country(name[:3]) + name[3:]
+                    )
+        else:
+            gdf_shapes = get_GADM_layer(co, gadm_level)
+            col = "GID_{}".format(gadm_level)
+    gdf_shapes = gdf_shapes[gdf_shapes[col].str.contains(co)]
+    return gdf_shapes, col
 
 
 def locate_bus(
-    coords,
-    co,
+    df,
+    countries,
     gadm_level,
     path_to_gadm=None,
     gadm_clustering=False,
-    col="name",
+    dropnull=True,
+    col_out=None,
 ):
     """
-    Function to locate the right node for a coordinate set input coords of
-    point.
+    Function to locate the points of the dataframe df into the GADM shapefile.
 
     Parameters
     ----------
-    coords: pandas dataseries
-        dataseries with 2 rows x & y representing the longitude and latitude
-    co: string (code for country where coords are MA Morocco)
-        code of the countries where the coordinates are
+    df: pd.Dataframe
+        Dataframe with mandatory x, y and country columns
+    countries: list
+        List of target countries
+    gadm_level: int
+        GADM level to be used
+    path_to_gadm: str (default None)
+        Path to the GADM shapefile
+    gadm_clustering: bool (default False)
+        True if gadm clustering is adopted
+    dropnull: bool (default True)
+        True if the rows with null values should be dropped
+    col_out: str (default gadm_{gadm_level})
+        Name of the output column
     """
-    col = "name"
-    if not gadm_clustering:
-        gdf = gpd.read_file(path_to_gadm)
-    else:
-        if path_to_gadm:
-            gdf = gpd.read_file(path_to_gadm)
-            if "GADM_ID" in gdf.columns:
-                col = "GADM_ID"
+    if col_out is None:
+        col_out = "gadm_{}".format(gadm_level)
+    df = df[df.country.isin(countries)]
+    df[col_out] = None
+    for co in countries:
+        gdf_shape, col = _get_shape_col_gdf(
```

Comment:
I recognise that it may be opening Pandora box, but wouldn't it be possible to replace naming for `df` and `col` with something a bit more specific? To my perception, it could probably facilitate a bit reading of the code. What do you think?

Reply:
Ok, I'll try :)